### ISSUE TITLE
Implement training function for Unet class

### ISSUE THREAD

**Issue by JacksonMHoke** [JacksonMHoke/diffusion_project#8]
Implement training function for Unet class that takes in optimizer, loss function, epochs, etc
